### ISSUE TITLE
How to close or get rid of blank page which consumes unused memory

### ISSUE THREAD

**Issue by praveenkutty** [thomasdondorf/puppeteer-cluster#241]
Hey,
when I tried your example code in headless:false mode it opens a blank page which is idle and stays until the task are completed. And if the task goes on for a long time this blank page is observed to use around 600 Mb.  How can I get rid of that blank page since it consumes some memory ?

Here is your example code which reproduces this issue

`const { Cluster } = require('../dist');

(async () => {
    // Create a cluster with 2 workers
    const cluster = await Cluster.launch({
  puppeteerOptions: {
            headless: false,
            },
        concurrency: Cluster.CONCURRENCY_CONTEXT,
        maxConcurrency: 2,
    });

    // Define a task (in this case: screenshot of page)
    await cluster.task(async ({ page, data: url }) => {
        await page.goto(url);

        const path = url.replace(/[^a-zA-Z]/g, '_') + '.png';
        await page.screenshot({ path });
        console.log(`Screenshot of ${url} saved: ${path}`);
    });

    // Add some pages to queue
    cluster.queue('https://www.google.com');
    cluster.queue('https://www.wikipedia.org');
    cluster.queue('https://github.com/');

    // Shutdown after everything is done
    await cluster.idle();
    await cluster.close();
})();`

**Comment by thomasdondorf** [comment]
Yes, good point. This part could be improved by using the first (already open) page instead of opening another one...

**Comment by quetzalsly** [comment]
Please don't do this because then puppeteer-extra will not work properly! They only apply hooks on new pages not existing ones.

**Comment by praveenkutty** [comment]
@thomasdondorf ,Thanks for your reply

With puppeteer I was able to isolate the blank page instance using below ,

```
const browser = await puppeteer.launch()
const page = (await browser.pages())[0];
await page.goto('url', { waitUntil: [ 'domcontentloaded', 'load']  });
```
But the same code does not work in puppeteer-cluster, I am sure am missing something here
Is there any way to get the blank page instance in puppeteer cluster?

May be like::
```
    let cluster = await getBrowser();
    await cluster.task(async ({ (cluster.pages())[0], data: url }) => {
             //code 
        }
```

**Comment by thomasdondorf** [comment]
@quetzalsly Thanks for the hint. As I already know of several people using that, I'm not going to change it...

@praveenkutty No, there is no simple way to do this. Your task function only get the page and not the complete browser object.

**Comment by edi2004** [comment]
I have some workaround to prevent the unused blank page.
I've just use the page (window) by using the page.browser() function, using the script below:

```
const { Cluster } = require('puppeteer-cluster')

;(async () => {
    const cluster = await Cluster.launch({
        concurrency: Cluster.CONCURRENCY_CONTEXT,
        maxConcurrency: 1,
        puppeteerOptions: {
            headless: false,
            devtools: false
        }
    })

    cluster.queue(async ({ page }) => {
        const browser = page.browser()
        const pagesArray = await browser.pages()
        console.log(pagesArray.length)
        const page2 = pagesArray[0]
        await page2.goto('http://www.wikipedia.org')
        await page2.screenshot({path: 'wikipedia.png'})
    })

    cluster.queue(async ({ page }) => {
        await page.goto('https://www.google.com/')
        const pageTitle = await page.evaluate(() => document.title)
        console.log(`Page title is ${pageTitle}`)
    })

    cluster.queue(async ({ page }) => {
        await page.goto('https://www.google.com/')
    })

    // await cluster.idle()
    // await cluster.close()
})()
```

As you can see the result of `console.log(pagesArray.length)` is 2
